### PR TITLE
feat(searchbar): add disabled attribute

### PR DESCRIPTION
**Pull request by KyDenZ**
#### Short description of what this resolves:
It is now possible to disable the searchbar component

#### Changes proposed in this pull request:
- New attribute disabled on searchbar component

**Ionic Version**: 4.1.2

**Fixes**: #17921

### CODE REVIEW

**Comment by brandyscarney** [comment]
Thank you for the PR! The code looks good! One question, why was the  `ionStyle` and `emitStyle` added? I'm guessing it was copied from another component. 🙂 

I don't think we need these as they are used to pass classes up to a parent component for styling and I don't see toolbar using them. Please correct me if I'm mistaken. :grin:

**Comment by KyDenZ** [comment]
Indeed, well seen 🙂
I relied on the "input" component. I thought it was used to re-type the style after a status change disabled or not.

File: core/src/components/searchbar/searchbar.tsx [hunk 1]
```diff
@@ -78,6 +78,16 @@ export class Searchbar implements ComponentInterface {
     this.ionChange = debounceEvent(this.ionChange, this.debounce);
   }
 
+  /**
+   * If `true`, the user cannot interact with the input.
+   */
+  @Prop() disabled = false;
+
+  @Watch('disabled')
```
Comment (brandyscarney) [inline core/src/components/searchbar/searchbar.tsx]:
```suggestion

```

File: core/src/components/searchbar/searchbar.tsx [hunk 2]
```diff
@@ -78,6 +78,16 @@ export class Searchbar implements ComponentInterface {
...
+  protected disabledChanged() {
```
Comment (brandyscarney) [inline core/src/components/searchbar/searchbar.tsx]:
```suggestion

```

File: core/src/components/searchbar/searchbar.tsx [hunk 3]
```diff
@@ -78,6 +78,16 @@ export class Searchbar implements ComponentInterface {
...
+    this.emitStyle();
```
Comment (brandyscarney) [inline core/src/components/searchbar/searchbar.tsx]:
```suggestion

```

File: core/src/components/searchbar/searchbar.tsx [hunk 4]
```diff
@@ -78,6 +78,16 @@ export class Searchbar implements ComponentInterface {
...
+  }
```
Comment (brandyscarney) [inline core/src/components/searchbar/searchbar.tsx]:
```suggestion

```

File: core/src/components/searchbar/searchbar.tsx [hunk 5]
```diff
@@ -148,6 +158,12 @@ export class Searchbar implements ComponentInterface {
     this.ionChange.emit({ value });
   }
 
+  /**
+   * Emitted when the styles change.
+   * @internal
+   */
+  @Event() ionStyle!: EventEmitter<StyleEventDetail>;
```
Comment (brandyscarney) [inline core/src/components/searchbar/searchbar.tsx]: Please remove this

File: core/src/components/searchbar/searchbar.tsx [hunk 6]
```diff
@@ -346,15 +362,30 @@ export class Searchbar implements ComponentInterface {
     return this.value || '';
   }
 
+  private hasValue(): boolean {
+    return this.getValue() !== '';
+  }
+
+  private emitStyle() {
```
Comment (brandyscarney) [inline core/src/components/searchbar/searchbar.tsx]: Please remove this

File: core/src/components/searchbar/searchbar.tsx [hunk 7]
```diff
@@ -346,15 +362,30 @@ export class Searchbar implements ComponentInterface {
...
+      'searchbar-has-focus':  this.focused,
+      'interactive-disabled': this.disabled
+    });
+  }
+
   hostData() {
     const animated = this.animated && this.config.getBoolean('animated', true);
 
     return {
+      'searchbar-disabled': this.disabled ? 'true' : null,
```
Comment (brandyscarney) [inline core/src/components/searchbar/searchbar.tsx]:
```suggestion
      'aria-disabled': this.disabled ? 'true' : null,
```

File: core/src/components/searchbar/searchbar.tsx [hunk 8]
```diff
@@ -346,15 +362,30 @@ export class Searchbar implements ComponentInterface {
...
       class: {
         ...createColorClasses(this.color),
         'searchbar-animated': animated,
```
Comment (brandyscarney) [inline core/src/components/searchbar/searchbar.tsx]:
```suggestion
        'searchbar-animated': animated,
        'searchbar-disabled': this.disabled,
```

Comment (brandyscarney) [reply core/src/components/searchbar/searchbar.tsx]: I don't know why it's showing animated as a new line but that one shouldn't change :grin:

**Comment by KyDenZ** [comment]
Thank you for your changes 🙂

**Comment by KyDenZ** [comment]
What are the steps to validate the feature? 🙂

**Comment by brandyscarney** [comment]
@KyDenZ Sorry I still need to take another look at this. I've been back and forth on PRs and getting `4.3.0` released. I was going to run it locally and verify the UI + I need to check out the pointer events.

**Comment by brandyscarney** [comment]
Moved the opacity and pointer-events up to the main searchbar so the icon will appear disabled. This looks good! Thank you!!

Screen Shot 2019-04-17 at 5 40 08 PM
Screen Shot 2019-04-17 at 5 40 18 PM

@liamdebeasi Could you add your stamp of approval? Ben marked off on the design.

**Comment by KyDenZ** [comment]
Thank you 🙂

**Comment by brandyscarney** [comment]
Thanks! :tada:

**Pull request merged**